common/api/core-geometry.api.md

```diff
@@ -4016,7 +4016,7 @@ export class Point3d extends XYZ {
     static fromJSON(json?: XYZProps): Point3d;
     interpolate(fraction: number, other: XYAndZ, result?: Point3d): Point3d;
     interpolatePerpendicularXY(fraction: number, pointB: Point3d, fractionXYPerp: number, result?: Point3d): Point3d;
-    interpolatePointAndTangent(fraction: number, other: Point3d, tangentScale: number, result?: Ray3d): Ray3d;
+    interpolatePointAndTangent(fraction: number, other: Point3d, tangentScale: number, result: Ray3d): Ray3d;
```

Comment:
result is supposed to be optional. You only supply this when you have an existing Ray3d you want to update, you leave it undefined to return a new Ray3d. Lots of geom apis follow this pattern...

Reply:
I see the motivation, but this feels funky and really easy to backslide on. Any new methods in Point3d that created Ray3d would reintroduce the dependency... it doesn't feel worth a breaking change to get a fragile win.

Reply:
I realize this is inconsistent, but was hoping this would be the least disruptive way to remove that hard dependency.

The alternative here would have to be moving this method out of the Point3d class...

Reply:
and @mgooding, I agree about the risk of backsliding here - that's why I changed the `Ray3d` import to `import type`.  The hope was that will help us catch that in a review in the future.  We should also probably have some automated tools regularly measuring bundle sizes of various extensions.

But it's worth emphasizing how important this is to our plan w/ extensions - without this change my simple example extension (the core select tool) was >57% geometry code.  Even if runtime extensions are lazily loaded, I think we still can't really afford that much waste/overhead.

Reply:
An obvious new home would be the Ray3d class: the equivalent function would take the form of a "setFrom" function that sets the instance Ray3d from two Point3ds and two numbers. Does deprecation allow extension savings, or would you have to remove the Point3d function altogether (breaking change)?

Reply:
I see no callers to this method. I could be moved to become a method on `Ray3d`

Reply:
I'd definitely prefer David's suggestion over this change.

Reply:
sorry @dassaf4, our messages crossed in the mail.

Reply:
@dassaf4 unfortunately deprecating on its own would not help us here... actually removing this from Point3d and moving it to Ray3d would work though.

A somewhat in-between option might be to make this change (result param mandatory) _plus_ deprecate the Point3d method, and then add it to Ray3d properly...  I'd probably lean towards "it's still a breaking change so what would even be the point?", but I suppose that might be a bit less disruptive than just removing the Point3d method.

Would the name `interpolatePointAndTangent` still be appropriate on `Ray3d`?

Reply:
@wgoehrig it's not used anywhere (at least by us). Let's just get rid of it pre-3.0 and document that it moved.

Reply:
Agree with @kabentley

Reply:
OK, just pushed that change - let me know what you think.